src/pages/index.js

```diff
@@ -21,91 +34,155 @@ popUpImageModal.setEventListeners();
 function handleCardClick(name, link) {
   popUpImageModal.openModal(name, link);
 }
-
+//Popup for deleting cards from API along with handler
+const dltPopup = new PopupWithForm(constants.selectors.deleteSelector);
+dltPopup.setEventListeners();
+function handleDelete(id, card) {
+  dltPopup.openModal();
+  dltPopup.setSubmitAction(() => {
+    api
+      .deleteCards(id)
+      .then(() => {
+        dltPopup.closeModal();
+        card.deleteUI();
+      })
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**